Wikipedia/Code/UIViewController+WMFOpenExternalUrl.h

```diff
@@ -19,4 +19,13 @@
  */
 - (void)wmf_openExternalUrl:(NSURL*)url useSafari:(BOOL)useSafari;
 
+/**
+ *  Open an external URL and specifiy whether or not safari should be used, and whether or not operating system version should be ignored
+ *
+ *  @param url                           The url to use
+ *  @param useSafari                     If YES, then safari will always be used
+ *  @param ignoreOperatingSystemVersion  If NO and it's iOS 9 or later, then safari will always be used
+ */
+- (void)wmf_openExternalUrl:(NSURL*)url useSafari:(BOOL)useSafari ignoreOperatingSystemVersion:(BOOL)ignoreOperatingSystemVersion;
```

Comment:
@huayu0723 we have second part of this task if you are interested. We would like to modify the behavior of our link opening to not use Safari at all unless we specify it explicitly. Since you are already working on link handling here, would you like to take on the bigger change?

If not, no worries we will merge this in and work on it separately, But if so, here is the other changes we would like to make to our link handling:

Refactor the internals of:
```
 - (void)wmf_openExternalUrl:(NSURL*)url;
```
This method should now always open within the app. In iOS 8, `SVModalWebViewController`. In iOS 9 and above, use `SFSafariViewController`

Refactor the internals of:
```
- (void)wmf_openExternalUrl:(NSURL*)url useSafari:(BOOL)useSafari;
```
If useSafari is NO, then the behavior is is identical to the method above:  Open the link within the app. In iOS 8, `SVModalWebViewController`. In iOS 9 and above, use `SFSafariViewController`
If useSafari is YES, then the url should be opened externally in Safari.

If you do make these changes, then I don't think you will need the 3rd method anymore since we never user the operating system to implicitly use Safari.

Thanks again for contributing!